genie-common/src/main/java/com/netflix/genie/common/dto/JobExecution.java

```diff
@@ -130,6 +135,15 @@ protected JobExecution(final Builder builder) {
         return Optional.ofNullable(this.memory);
     }
 
+    /**
+     * Get the archival status of job files.
+     *
+     * @return the archival status as Optional as it could be null
```

Comment:
you have the `private final` variable as `@NotNull` which doesn't line up with this comment or with the fact this returns and `Optional`

Reply:
Removing the @NotNull and sticking to the convention followed by other fields.